domains/client/block-builder/src/lib.rs

```diff
@@ -179,9 +179,24 @@ where
 
         if let Some(inherent_data) = maybe_inherent_data {
             let inherent_extrinsics = Self::create_inherents(parent_hash, &api, inherent_data)?;
-            // reverse and push the inherents so that order is maintained
-            for inherent_extrinsic in inherent_extrinsics.into_iter().rev() {
-                extrinsics.push_front(inherent_extrinsic)
+
+            // TODO: This is used to keep compatible with gemini-3h, remove before next network
+            //
+            // HACK: in gemini-3h, the domain inherent extrinsic order is changed in the ER that derived
+            // from the consensus block #168431, we have to follow this change in the client side to ensure
+            // every domain node that sync from genesis will produce the same ER and hence can successfully
+            // submit ER to exend the previous ER.
+            let maintain_runtime_inherent_extrinsic_order = parent_number >= 168430u32.into();
```

Comment:
You should also check that the chain is in fact Gemini 3h using genesis hash or something equivalent.

Reply:
Added code to determine if the node is running in gemini-3h by checking the genesis hash, and tested in gemini-3h.